### ISSUE TITLE
Adding observed parameters to a pre-trained model

### ISSUE THREAD

**Issue by aarcangeli** [Unity-Technologies/ml-agents#2016]
Hi, I have trained a model capable of performing a particular task.
It works pretty well.

Now I want to extend his skills so it can do a bigger task.
In order to do this I have to modify the algorithm and introduce new observations.
Is there a way to do this without re-training the model from scratch?

**Comment by ervteng** [comment]
> introduce new observations

There currently isn't a way to do this in ML-Agents, aside from highly manual Tensorflow operations (e.g. extracting the weights of the network and appending it to a subset of the new network).

**Comment by aarcangeli** [comment]
Thanks for the reply.
Maybe a solution is to add a 'padding' of fixed parameters for a possible future reuse.

For example, if my model has 300 values, I train a model of 350 values (where the last 50 are 0).
Whenever I need one more parameter, I replace one of the existing input with my observation.

Can it cause problems in training?

**Comment by ervteng** [comment]
Hi @aarcangeli, this should work, as long as the 350 values are a strict superset of the 300 and in the same positions. It shouldn't be a problem for training on the smaller set. But it's hard to predict what effect it will have when you transfer it to the new scenario - it should help if the 300 are the most important ones, but it really depends on the relationship that the network has learned. I'd give it a go!

**Comment by aarcangeli** [comment]
Thank you very much for the answers
